# Reproducing Paper's Output

## WeatherBench Results

In [16]:
from experimenter import get_exp_count
from inference import inference_on_test
from configs.weatherbench.default_conf import experiment_params


inference_params = {
    "start_date_str": "01-01-2017",
    "end_date_str": "01-01-2018",
    "test_data_folder": "data/weatherbench/test_data",
    # "exp_num": 2,  # or set it by yourself
    "forecast_horizon": 72,
    "selected_dim": -1  # index position of the selected feature
}
model_names = ['weather_model', 'u_net', 'traj_gru', 'convlstm']

### Sequential

In [12]:
# Performing sequential
from configs.weatherbench.seq_model_confs import model_params

exp_type = "sequential"
exp_dir = f"results/{exp_type}_results"
inference_params["exp_dir"] = exp_dir

In [13]:
# perform inference on test
for mn in model_names:
    inference_params["model_name"] = mn
    inference_params["exp_num"] = get_exp_count(mn, result_dir=exp_dir)  # get the last experiment
    inference_on_test(dataset_type="weatherbench", device=experiment_params["device"], **inference_params)

-*--*--*--*--*--*--*--*--*--*--*--*--*--*--*--*--*--*--*--*-
Inference on data/weatherbench/test_data between 01-01-2017 and 01-01-2018 dates
	test:15/16-*--*--*--*--*--*--*--*--*--*-
Experiment finished for the 01-01-2017_01-01-2018 the scores are: 
inference-test:		MSE: 23.41013, MAE: 3.68598, MAPE: 0.01349, RMSE: 4.83840, WeightedMAE: 3.25682, WeightedRMSE: 3.97436, WeightedACC: 0.69092, 
-*--*--*--*--*--*--*--*--*--*--*--*--*--*--*--*--*--*--*--*-
Inference on data/weatherbench/test_data between 01-01-2017 and 01-01-2018 dates
	test:15/16-*--*--*--*--*--*--*--*--*--*-
Experiment finished for the 01-01-2017_01-01-2018 the scores are: 
inference-test:		MSE: 11.04407, MAE: 2.28841, MAPE: 0.00842, RMSE: 3.32326, WeightedMAE: 2.07575, WeightedRMSE: 3.05444, WeightedACC: 0.82030, 
-*--*--*--*--*--*--*--*--*--*--*--*--*--*--*--*--*--*--*--*-
Inference on data/weatherbench/test_data between 01-01-2017 and 01-01-2018 dates
	test:15/16-*--*--*--*--*--*--*--*--*--*-
Experiment finished for th

### Iterative

In [17]:
from configs.weatherbench.iter_model_confs import model_params

exp_type = "iterative"
exp_dir = f"results/{exp_type}_results"
inference_params["exp_dir"] = exp_dir

In [18]:
# perform inference on test
for mn in model_names:
    print(mn)
    inference_params["model_name"] = mn
    inference_params["exp_num"] = get_exp_count(mn, result_dir=exp_dir)  # get the last experiment
    inference_on_test(dataset_type="weatherbench", device=experiment_params["device"], **inference_params)

weather_model
-*--*--*--*--*--*--*--*--*--*--*--*--*--*--*--*--*--*--*--*-
Inference on data/weatherbench/test_data between 01-01-2017 and 01-01-2018 dates
Performing iterative prediction since window_out (6) < forecast_horizon (72)
	test:3/4-*--*--*--*--*--*--*--*--*--*-
Experiment finished for the 01-01-2017_01-01-2018 the scores are: 
inference-test:		MSE: 11.92691, MAE: 2.36522, MAPE: 0.00877, RMSE: 3.45354, WeightedMAE: 2.08508, WeightedRMSE: 2.91743, WeightedACC: 0.83784, 
u_net
-*--*--*--*--*--*--*--*--*--*--*--*--*--*--*--*--*--*--*--*-
Inference on data/weatherbench/test_data between 01-01-2017 and 01-01-2018 dates
	test:15/16-*--*--*--*--*--*--*--*--*--*-
Experiment finished for the 01-01-2017_01-01-2018 the scores are: 
inference-test:		MSE: 22.18061, MAE: 3.63348, MAPE: 0.01341, RMSE: 4.70963, WeightedMAE: 3.26551, WeightedRMSE: 4.13053, WeightedACC: 0.68033, 
traj_gru
-*--*--*--*--*--*--*--*--*--*--*--*--*--*--*--*--*--*--*--*-
Inference on data/weatherbench/test_data betw

### Direct

In [9]:
from configs.weatherbench.direct_model_confs import model_params

model_name = "weather_model"  # can be ['weather_model', 'u_net', 'traj_gru', 'convlstm']
exp_type = "direct"

In [10]:
# perform inference on test
exp_dir = f"results/{exp_type}_results"
inference_params = {
    "model_name": model_name,
    "start_date_str": "01-01-2017",
    "end_date_str": "01-01-2018",
    "test_data_folder": "data/weatherbench/test_data",
    "exp_dir": exp_dir,
    "exp_num": get_exp_count(model_name, result_dir=exp_dir),  # get the last experiment
    # "exp_num": 2,  # or set it by yourself
    "forecast_horizon": 72,
    "selected_dim": -1  # index position of the selected feature
}
inference_on_test(dataset_type="weatherbench", device=experiment_params["device"], **inference_params)

FileNotFoundError: [Errno 2] No such file or directory: 'results/direct_results/weather_model/exp_0/model.pkl'

### Results

| Model Name             | WeightedRMSE   | WeightedMAE   | WeightedACC |
|------------------------|----------------|---------------|-------------|
| Convlstm (seq)         | 4.43           | 3.47          | 0.64        |
| TrajGRU (seq)          | 3.598          | 2.79          | 0.76        |
| U-Net (seq)            | **3.06**       | **2.08**      | **0.82**    |
| Weather Model (seq)    | 3.97           | 3.26          | 0.69        |
| ---------------------  | -------------- | ------------- | ----------- |
| Convlstm (iter)        | 3.16           | 3.47          | 0.64        |
| TrajGRU (iter)         | 3.07           | 2.19          | 0.82        |
| U-Net (iter)           | 4.13           | 3.26          | 0.68        |
| Weather Model (iter)   | **2.92**       | **2.08**      | **0.84**    |
| ---------------------  | -------------- | ------------- | ----------- |
| Weather Model (direct) | 3.07           | 2.20          | 0.82        |